blockmanager.go

```diff
@@ -774,15 +774,17 @@ func (b *blockManager) getUncheckpointedCFHeaders(
 				"headers", len(badPeers))
 
 			for _, peer := range badPeers {
-				log.Infof("Banning peer=%v for invalid filter "+
-					"headers", peer)
-
+				err := b.server.BanPeer(
+					peer, banman.InvalidFilterHeader,
```

Comment:
I noticed that peers that don't respond will still be banned for the `InvalidFilterHeader` reason here which could be a little misleading.

Reply:
Seems like we'd have to return a ban reason from `detectBadPeers` since there are different scenarios that can cause them to get banned. I think that's fine in this case though because they previously did respond with what could've been an invalid filter header.